disco/cli/compute_pv_swad.py

```diff
@@ -0,0 +1,156 @@
+
+    return total_existing_pv, existing_pv_dict, existing_pv_dist_dict
+
+
+def compute_weighted_average_pv_distance(load_info, option="size"):
+    wavg_pv_distance = 0
+    total_existing_pv, existing_pv_dict, existing_pv_dist_dict = get_existing_pvs()
+
+    slack_dico = {k: v * existing_pv_dist_dict[k] for k, v in existing_pv_dict.items()}
+
+    if option == "size":
+        wavg_pv_distance = sum(slack_dico.values()) / max(0.001, total_existing_pv)
+
+    elif option == "nodal_ratio":
+        load_per_bus = {v["bus"]: v["kW"] for k, v in load_info.items()}
+
+        slack_dico = {
+            k: v * existing_pv_dist_dict[k] / max(0.001, load_per_bus[k])
+            for k, v in existing_pv_dict.items()
+        }
+        ratio_sum = sum([v / max(0.001, load_per_bus[k]) for k, v in existing_pv_dict.items()])
+        wavg_pv_distance = sum(slack_dico.values()) / max(0.001, ratio_sum)
+
+    return wavg_pv_distance
+
+
+def collect_weighted_average_pv_distance(filename: Path, option):
+    reply = dss.utils.run_command(f"compile {filename}")
+    if reply != "":
+        print(f"Failed to compile OpenDSS model {filename}: {reply}", file=sys.stderr)
+        sys.exit(1)
+
+    load_info = get_load_info()
+    avg = compute_weighted_average_pv_distance(load_info, option=option)
+    print(f"Computed weighted average PV distance {avg} for {filename}")
+    return avg
+
+
+def collect_weighted_average_pv_distances(path: Path, option):
+    results = []
+    for substation_path in path.iterdir():
+        if substation_path.is_dir():
+            for feeder_path in substation_path.iterdir():
+                pv_path = feeder_path / "PVDeployments"
+                if pv_path.is_dir():
+                    for filename in pv_path.iterdir():
+                        job_name = filename.name.replace(".dss", "")
+                        fields = job_name.split("__")
+                        if len(fields) != 5:
+                            continue
+                        substation, feeder, placement, sample, penetration_level = fields
+                        wavg_pv_distance = collect_weighted_average_pv_distance(filename, option)
+                        results.append(
+                            {
+                                "job_name": job_name,
+                                "substation": substation,
+                                "feeder": feeder,
+                                "placement": placement,
+                                "sample": sample,
+                                "penetration_level": penetration_level,
+                                "weighted_average_pv_distance": wavg_pv_distance,
+                                "option": option,
+                            }
+                        )
+    return results
+
+
+def main():
+    if len(sys.argv) == 1:
+        print(
+            f"Usage: python {sys.argv[0]} PV_DEPLOYMENTS_PATH [size|nodal_ratio]", file=sys.stderr
+        )
```

Comment:
Not sure if it's worth a refactoring, a `click.command` may be more user friendly.